src/main/java/com/mongodb/hibernate/jdbc/MongoPreparedStatement.java

```diff
@@ -187,9 +190,20 @@ public void setTimestamp(int parameterIndex, Timestamp x) throws SQLException {
 
     @Override
     public void setObject(int parameterIndex, Object x, int targetSqlType) throws SQLException {
+        // VAKOTODO add tests
         checkClosed();
         checkParameterIndex(parameterIndex);
-        throw new FeatureNotSupportedException("To be implemented in scope of Array / Struct tickets");
+        BsonValue value;
+        if (targetSqlType == MqlType.OBJECT_ID.getVendorTypeNumber()) {
+            if (x instanceof ObjectId v) {
+                value = new BsonObjectId(v);
+            } else {
+                throw fail();
+            }
+        } else {
+            throw new FeatureNotSupportedException("To be implemented in scope of Array / Struct tickets");
+        }
```

Comment:
It is a little bit pity that we can't use the JDK feature as below:
```
BsonValue value = switch(targetSqlType) {
    case MqlType.OBJECT_ID.getVendorTypeNumber() -> ...
    default -> ...
};
```
due to the fact that the vendor type number is not a constant.